### PR TITLE
Update matrix-domain.conf.j2 - trying to fix issue #2954

### PR DESCRIPTION
**Pull request by jolly-jump**
exchanged `^~` with `~` as a pattern matching in the location part. I am very sure, that it only works using `~`. I am not quite sure though, if this is the right way to do it, because `~` is probably more expensive than `^~`

the rewrite has to be behind the definition of the $backend. Otherwise nginx will fail to work. This is probably because "break" goes directly to the proxy_pass which uses $backend.

### CODE REVIEW

**Comment by spantaleev** [comment]
I suppose we need the same `if` (and `rewrite` block) once again a bit below then, to cover the `else` statement?

File: roles/custom/matrix-nginx-proxy/templates/nginx/conf.d/matrix-domain.conf.j2 [hunk 1]
```diff
@@ -207,16 +209,18 @@
...
 		{% if matrix_nginx_proxy_enabled %}
 			{# Use the embedded DNS resolver in Docker containers to discover the service #}
 			resolver {{ matrix_nginx_proxy_http_level_resolver }} valid=5s;
 			set $backend "{{ matrix_nginx_proxy_proxy_matrix_3pid_registration_addr_with_container }}";
+   		    {% if matrix_nginx_proxy_proxy_matrix_3pid_registration_v3_to_r0_redirect_enabled %}
+            rewrite ^(.*?)/v3/(.*?)$  $1/r0/$2 break;
+	        {% endif %}
 			proxy_pass http://$backend;
 		{% else %}
+            {% if matrix_nginx_proxy_proxy_matrix_user_directory_search_v3_to_r0_redirect_enabled %}
```
Comment (jolly-jump) [inline roles/custom/matrix-nginx-proxy/templates/nginx/conf.d/matrix-domain.conf.j2:221]: mistake, will fix

**Comment by jolly-jump** [comment]
Hi. So I did what you suggested.
This PR works now as expected for user_directory search.
Do you want me to rebase and make a new PR?
Pretty messy six commits for the small change :shrug: ?

**Comment by spantaleev** [comment]
Thanks for fixing it! I've squashed it, so it's clean enough now.

**Pull request merged**